core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -702,6 +735,42 @@ public Builder version(long version) {
             return this;
         }
 
+        /**
+         * returns the primary term for the given shard.
+         * See {@link IndexMetaData#primaryTerm(int)} for more information.
+         */
+        public long primaryTerm(int shardId) {
+            if (primaryTerms == null) {
+                initializePrimaryTerms();
+            }
+            return this.primaryTerms[shardId];
+        }
+
+        /**
+         * sets the primary term for the given shard.
+         * See {@link IndexMetaData#primaryTerm(int)} for more information.
+         */
+        public Builder primaryTerm(int shardId, long primaryTerm) {
+            if (primaryTerms == null) {
+                initializePrimaryTerms();
+            }
+            this.primaryTerms[shardId] = primaryTerm;
+            return this;
+        }
+
+        private void primaryTerms(long[] primaryTerms) {
+            this.primaryTerms = primaryTerms;
```

Comment:
You didn’t missread - the only thing is that the diffs are read of the network and are discarded. That one pulled me over the line to actually change it and copy the array.

> On 21 Oct 2015, at 18:15, Jason Tedor <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java:
> 
> > +        }
> > +
> > +        /**
> > +         * sets the primary term for the given shard.
> > +         * See {@link IndexMetaData#primaryTerm(int)} for more information.
> > +         */
> > +        public Builder primaryTerm(int shardId, long primaryTerm) {
> > +            if (primaryTerms == null) {
> > +                initializePrimaryTerms();
> > +            }
> > +            this.primaryTerms[shardId] = primaryTerm;
> > +            return this;
> > +        }
> > +
> > +        private void primaryTerms(long[] primaryTerms) {
> > +            this.primaryTerms = primaryTerms;
> 
> Maybe I misread, but I think there's one place where it's not in IndexMetaDataDiff.apply?
> 
> —
> Reply to this email directly or view it on GitHub.
>